src/plugins/controls/public/options_list/components/options_list_editor_options.tsx

```diff
@@ -6,43 +6,96 @@
  * Side Public License, v 1.
  */
 
-import React, { useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
 import {
   EuiFlexGroup,
   EuiFlexItem,
+  EuiForm,
   EuiFormRow,
   EuiIconTip,
+  EuiSuperSelectOption,
+  EuiSpacer,
+  EuiSuperSelect,
   EuiSwitch,
   EuiSwitchEvent,
+  EuiButtonGroup,
+  toSentenceCase,
+  Direction,
 } from '@elastic/eui';
 import { css } from '@emotion/react';
 
+import {
+  getCompatibleSortingTypes,
+  sortDirections,
+  DEFAULT_SORT,
+  SortBy,
+} from '../../../common/options_list/suggestions_sorting';
 import { OptionsListStrings } from './options_list_strings';
 import { ControlEditorProps, OptionsListEmbeddableInput } from '../..';
+
 interface OptionsListEditorState {
-  singleSelect?: boolean;
+  sortDirection: Direction;
   runPastTimeout?: boolean;
+  singleSelect?: boolean;
   hideExclude?: boolean;
   hideExists?: boolean;
+  hideSort?: boolean;
+  sortBy: SortBy;
```

Comment:
Nit: Not sure how other folks feel about this, but I like to preface most types with a little context - like `OptionsListSortBy` for instance. It makes the names longer which can be annoying to type, but that way we never confuse our system types for more generic / fundamental ones offered by other teams.

Reply:
Yeah that's a good point! Since it's specific to options list (at least for now), I think it's a good idea to make the name reflect this 👍